src/test/java/com/melon/portfoliomanager/UserControllerTest.java

```diff
@@ -0,0 +1,130 @@
+package com.melon.portfoliomanager;
+
+import com.melon.portfoliomanager.dtos.UserDeleteDto;
+import com.melon.portfoliomanager.dtos.UserDto;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.repositories.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.web.reactive.function.BodyInserters;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserControllerTest {
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    private UserDto validUserDto;
+
+    @BeforeEach
+    void setUp() {
+        validUserDto = new UserDto();
+        validUserDto.setUsername("vvp");
+        validUserDto.setEmail("vpavlov@melon.com");
+        validUserDto.setFirstName("Georgi");
+        validUserDto.setLastName("Ivanov");
+    }
+
+    @Test
+    void createUser_ValidRequest_ShouldReturnCreated() {
+
+        when(userRepository.save(any(User.class))).thenReturn(new User("vvp", "vpavlov@melon.com", "Georgi", "Ivanov"));
+
+        webTestClient.post().uri("/users/add")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(validUserDto))
+                .exchange()
+                .expectStatus().isCreated();
+    }
+
+    @Test
+    void createUser_InvalidRequest_WrongEmail_ShouldReturnBadRequest() {
+        webTestClient.post().uri("/users/add")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(new UserDto("georgi", null, "v", "p")))
+                .exchange()
+                .expectStatus().isBadRequest();
+    }
+
+    @Test
```

Comment:
It is best to have also test that validates that when unexpected error occurs then our endpoints react properly and log useful messages.